updatemanager.go

```diff
@@ -138,6 +138,10 @@ func newUpdateManager(cfg *config.Config) (um *updateManager, err error) {
 }
 
 func (um *updateManager) close() {
+	if um == nil {
```

Comment:
I prefer fix in newUpdateManager: move defer function after assigning um variable. The point is: we should not call close for not created object.